client/app/dedicatedCloud/security/dedicatedCloud-security.controller.js

```diff
@@ -136,5 +154,33 @@ angular.module('App').controller('DedicatedCloudSecurityCtrl', function ($rootSc
     $scope.setMessage($translate.instant('dedicatedCloud_dashboard_loading_error'), data.data);
   };
 
+  self.addKms = function () {
+    const kmsCreationModal = $uibModal.open({
+      templateUrl: 'dedicatedCloud/security/kms/add/dedicatedCloud-security-kms-add.html',
+      controller: 'DedicatedCloudSecurityKMSAddCtrl',
+      controllerAs: '$ctrl',
+    });
+
+    kmsCreationModal.result.then(() => {
+      $scope.loadInfo();
+      $scope.setMessage($translate.instant('dedicatedCloud_vm_encryption_kms_added'), {});
+    });
+  };
+
+  self.getVMEncryptionKMSList = function ({ offset, pageSize }) {
+    return DedicatedCloud.getVMEncryptionKMSList($stateParams.productId).then(kmsIds => ({
+      data: kmsIds
+        .slice(offset - 1, offset - 1 + pageSize)
+        .map(id => ({ id })),
+      meta: {
+        totalCount: kmsIds.length,
+      },
+    }));
+  };
+
+  self.getVMEncryptionKMSDetail = function ({ id }) {
```

Comment:
having an object here doesn't really make sense, you could you wait for the `id` and the caller will only pass the id.

I like to have objects in parameters when I have at least 3 parameters and I can't communicate in the name of the function what the parameters are.

For example `addIdIntoArray(id, array)` doesn't require an object